Feat padding column gap options

amanda-t opened the pull request:
[ENG-1534](https://linear.app/alephsf/issue/ENG-1534/add-additional-options-to-nought-sass-mixins-and-update-package). Adds support to set container padding and grid column gap at any breakpoint without deprecating `$container-padding-sm`. Or should I deprecate it and bump the package up a major version to 2.0.0? I'll run `standard-version` after this is merged in.

amanda-t committed:
fix: Add viewport meta tag

amanda-t committed:
feat: Allow for custom grid column gap & container padding across breakpoints

amanda-t committed:
fix: Remove unused vars

amanda-t committed:
docs: Add code comment

amanda-t committed:
feat: Add container-padding-sm back in to keep support

oppodeldoc reviewed (approved):
@amanda-t this looks good. Let's deprecate the `sm` option and use your code. Ready to merge and version!

amanda-t committed:
feat: Deprecate $container-padding-sm

BREAKING CHANGE: $container-padding-sm has been deprecated. Instead
use $container-responsive-padding to set the container padding for the
sm breakpoint as well as any other breakpoint you wish.
